test-case-reporting/app.ts

```diff
@@ -25,21 +25,22 @@ import statusCodes from 'http-status-codes';
 
 const MAX_PAGE_SIZE = 500;
 const DEFAULT_PAGE_SIZE = 100;
+const TEMPLATE_DIR = './static';
 
 const app = express();
 const jsonParser = bodyParser.json({limit: '15mb'});
 const db = dbConnect();
 const record = new TestResultRecord(db);
 
-let testRunListTemplate = '';
-function renderTestRunList(testRuns: Array<TestRun>): string {
-  if (!testRunListTemplate) {
-    testRunListTemplate = fs
-      .readFileSync('./static/test-run-list.html')
+const templateCache: Record<string, string> = {};
```

Comment:
Where does `Record` come from?

Reply:
It's a native TypeScript type

Reply:
Comes from TS -- https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt

Reply:
Oh neat. This might be useful if the types were non-basic but they're just strings. Otherwise I think the Map type gives you nice utilities like has, get and set.